webview_panels/src/modules/queryPanel/useQueryPanelListeners.ts

```diff
@@ -154,7 +158,24 @@ const useQueryPanelListeners = (): { loading: boolean } => {
     };
   }, [onMesssage]);
 
-  return { loading };
+  useEffect(() => {
+    if (!tabData) {
```

Comment:
Looks like you can avoid storing all the data in tabData. This is only needed to store whether this is a tab or not. You can just store this state value as boolean. This will avoid storing duplicate data in memory